src-tauri/src/app.rs

```diff
@@ -64,7 +65,7 @@ impl Context {
 
 pub struct Service {
     /// Handle onto the tauri application. The shared Context can be accessed and modified here.
-    context: Arc<Mutex<Context>>,
+    context: Arc<RwLock<Context>>,
```

Comment:
I've changed this from a `Mutex` to `RwLock`, it's an optimization and not necessary for this PR but I thought it makes sense as we are otherwise blocking access to other functions on every blob request.